cookbooks/mediawiki/templates/default/LocalSettings.php.erb

```diff
@@ -98,6 +98,23 @@ $wgFileExtensions[] = 'osm';
 	$wgFileExtensions[] = '<%= mw_extra_file_extension %>';
 <% end -%>
 
+# Add OSM XML file format per http://www.iana.org/assignments/media-types/media-types.xhtml
+# Shout out to Paul Norman for reserving this.
+$wgXMLMimeTypes[] = 'osm' => 'application/vnd.openstreetmap.data+xml';
+
+# Helps MimeMagic determine XML-based formats and chooses the correct MimeType
+# for .osm files.
+$wgXMLMimeTypes = array(
+	'http://www.w3.org/2000/svg:svg' => 'image/svg+xml',
+	'svg' => 'image/svg+xml',
+	'http://www.lysator.liu.se/~alla/dia/:diagram' => 'application/x-dia-diagram',
+	'http://www.w3.org/1999/xhtml:html' => 'text/html',
+	'html' => 'text/html',
+	'osm' => 'application/vnd.openstreetmap.data+xml' );
```

Comment:
What's the logic here? It looks like this is doing a complete replacement of the hash which is not really ideal as it means we won't get any future updates to the default value, and in any case why would we add a new entry and then replace the whole thing?

Reply:
This is correct - it is replacing the array. I will remove this and fix the above entry for $wgXMLMimeTypes to add the elements to the array dynamically.